packages/server/src/analytics/createDashboard.ts

```diff
@@ -0,0 +1,151 @@
+import { DashboardsServiceClient } from '@google-cloud/monitoring-dashboards'
+import { processEnvOrThrow, implementedStates, getStateAbbr } from '../common'
+import { AnalyticsStorage } from './storage'
+import { FirestoreService } from '../service/firestore'
+
+const fsService = new FirestoreService()
+const projectId = processEnvOrThrow('GCLOUD_PROJECT')
+const client = new DashboardsServiceClient()
+const projectPath = client.projectPath(projectId)
+
+// JSON models from this file are based on the documentation from:
+// https://cloud.google.com/monitoring/dashboards/api-examples#dashboard_in_gridlayout
+// https://cloud.google.com/monitoring/dashboards/api-dashboard
+// and the Metric Explorer https://console.cloud.google.com/monitoring/metrics-explorer
+
+const makeDataSet = (resourceUrl: string, label: string) => ({
+  plotType: 'LINE',
+  legendTemplate: label,
+  minAlignmentPeriod: { seconds: 10800 },
+  timeSeriesQuery: {
+    unitOverride: null,
+    timeSeriesFilter: {
+      aggregation: {
+        groupByFields: null,
+        alignmentPeriod: { seconds: 10800 },
+        crossSeriesReducer: 'REDUCE_NONE',
+        perSeriesAligner: 'ALIGN_MAX',
+      },
+      secondaryAggregation: {
+        groupByFields: null,
+        alignmentPeriod: null,
+        crossSeriesReducer: 'REDUCE_NONE',
+        perSeriesAligner: 'ALIGN_MAX',
+      },
+      filter: `metric.type="custom.googleapis.com/${resourceUrl}"`
+    }
+  }
+} as const)
+
+const makeWidget = (
+  widgetTitle: string,
+  yAxisLabel: null | string,
+  chartData: Array<{resourceUri: string, label: string} | null>,
+) => ({
+  title: widgetTitle,
+  xyChart: {
+    chartOptions: { mode: 'COLOR' },
+    thresholds: null,
+    timeshiftDuration: { seconds: 0 },
+    xAxis: null,
+    yAxis: { label: yAxisLabel, scale: 'LINEAR' },
+
+    dataSets: chartData.map(
+      d => d ? makeDataSet(d.resourceUri, d.label) : null
+    // Removes null elements and type casting errors
+    ).filter(e => !!e) as ReturnType<typeof makeDataSet>[],
+  }
+} as const)
+
+// We used to store this in analytics/storage, but apparently it can change
+// by itself in some occasions (we need this value to update dashboards).
+const getETag = async (dashboardName: string | null | undefined) => {
```

Comment:
If it works, this generally looks good to me, but would you mind adding a few more comments? Like, what is an eTag?